firmware/tools/make_spi_image.py

```diff
@@ -135,8 +135,8 @@ def get_gcc_version_from_elf_files_in_giving_path_or_filename_s_path(path):
     padded_data = image['data'] + (spi_image_default_byte * pad_size)
     spi_image += padded_data
 
-if len(spi_image) > spi_size - 4:
-    raise RuntimeError('SPI flash image size of %d exceeds device size of %d bytes' % (len(spi_image) + 4, spi_size))
+# if len(spi_image) > spi_size - 4:
+  #  raise RuntimeError('SPI flash image size of %d exceeds device size of %d bytes' % (len(spi_image) + 4, spi_size))
```

Comment:
In my opinion the build needs to fail when the image doesn't fit in the ROM.  Perhaps the message string should be changed to indicate how many bytes are in excess though.

Reply:
Agreed, it should be the default behaviour to fail, however we could have an environment variable controlling if it fails so that when some1 wants to test build succeed even in resulting in a too large image he could. I don't see the use anyway, if you see this error your build succeeded basically I'd guess that should be enough information, the image would be useless anyway.

Reply:
Let's go with an env car to disable the test, and a better print message then !

Reply:
Thanks to all of you for the above comme ts and revision .
Then so far , I will change back the 1MB limitation .
After Bernd revision , I will un-draft it .
Cheers